businesskaniki/nairobi-north
# Fetching Data from Api

# Pull Request: Actions and Reducers for Data Entities

## Description

This pull request introduces actions and reducers for managing various data entities in our application, including churches, ministries, prayer requests, videos, images, events, and sermons. It follows a consistent pattern with our existing actions and reducers for other data entities.

## Changes Made

1. Added actions for each data entity in their respective action files (`actions/churches.js`, `actions/ministries.js`, `actions/prayerRequests.js`, `actions/videos.js`, `actions/images.js`, `actions/events.js`, `actions/sermons.js`). These actions include fetching data, adding, editing, and deleting data for each entity.

2. Created corresponding reducers for each data entity in their respective reducer files (`reducers/churches.js`, `reducers/ministries.js`, `reducers/prayerRequests.js`, `reducers/videos.js`, `reducers/images.js`, `reducers/events.js`, `reducers/sermons.js`). These reducers manage the state related to each data entity, including loading, error handling, and storing data.

3. Utilized the Redux Toolkit `createAsyncThunk` for handling asynchronous actions, making it easy to fetch and update data for each entity.

## How to Test

You can test these changes by:

1. Dispatching the appropriate actions to fetch data for each data entity.
2. Dispatching actions to add, edit, and delete data for each entity.
3. Verifying that data is correctly fetched and displayed in the UI, and that updates and deletions are handled properly.

Please review and provide feedback on these changes.

## Note

This PR adheres to the Don't Repeat Yourself (DRY) principle by reusing the `apiRequest` function for making API requests. This ensures consistency in authentication and error handling for all data entities.
